edk2toollib/database/tables/source_table.py

```diff
@@ -0,0 +1,72 @@
+# @file source_table.py
+# A module to Parse all Source files and add them to the database.
+##
+# Copyright (c) Microsoft Corporation
+#
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+"""A module to Parse all Source files and add them to the database."""
+import logging
+import re
+import time
+from joblib import Parallel, delayed
+from pathlib import Path
+from tinyrecord import transaction
+from edk2toollib.database import Edk2DB, TableGenerator
+
+
+class SourceTable(TableGenerator):
+    """A Table Generator that parses all c and h files in the workspace.
+
+    Generates a table with the following schema:
+
+    table_name: "source"
+    |-------------------------------------------------------------------------|
+    | PATH | LICENSE | TOTAL_LINES | CODE_LINES | COMMENT_LINES | BLANK_LINES |
+    |-------------------------------------------------------------------------|
+    """  # noqa: E501
+    def __init__(self, *args, **kwargs):
+        """Initializes the Source Table Parser.
+
+        Args:
+            args (any): non-keyword arguments
+            kwargs (any): keyword arguments described below
+
+        Keyword Arguments:
+            n_jobs (int): Number of files to run in parallel
+        """
+        self.n_jobs = kwargs.get("n_jobs", -1)
+
+    def parse(self, db: Edk2DB, env: dict) -> None:
+        """Parse the workspace and update the database."""
+        ws = Path(db.pathobj.WorkspacePath)
+        src_table = db.table("source", cache_size=None)
+
+        start = time.time()
+        files = list(ws.rglob("*.c")) + list(ws.rglob("*.h"))
```

Comment:
Should this include cpp now that they are used for googletest? 

Also is assembly intentionally excluded?